crates/locutus-node/src/operations/get.rs

```diff
@@ -345,31 +350,62 @@ where
             fetch_contract,
             sender,
             target,
-            ..
+            htl,
         } => {
             if !op_storage.ring.is_contract_cached(&key) {
-                //FIXME: should try forward to someone else who may have it first
-                // this node does not have the contract, return a void result to the requester
                 log::warn!(
-                    "Contract `{}` not found while processing a get request",
-                    key
+                    "Contract `{}` not found while processing a get request at node @ {}",
+                    key,
+                    target.peer
+                );
+
+                if htl == 0 {
```

Comment:
wouldn't it be more logical to check if the new_htl == 0 after failing and avoiding one additional network message before returning to the initial node?

Reply:
my mistake, this is only checked IF the contract is not cached, but any other cases would be valid so the additional hop makes sense